doc/lsst.pipe.base/command-line-task-parallel-howto.rst

```diff
@@ -0,0 +1,97 @@
+.. _command-line-task-parallel-howto:
+
+###########################################
+Parallel processing with command-line tasks
+###########################################
+
+:ref:`Command-line tasks <using-command-line-tasks>` can operate in parallel.
+Generally if a command-line task is run against several :ref:`data IDs <command-line-task-dataid-howto>`, each data ID can be processed in parallel.
+Command-line tasks use two different categories of parallel processing: single-node parallel processing (:option:`-j`) and distributed processing (with the ``pipe_drivers`` and ``ctrl_pool`` packages).
+This page describes how to use these parallel processing features.
+
+.. _command-line-task-parallel-howto-multiprocessing:
+
+How to enable single-node parallel processing
+=============================================
+
+Command-line tasks can process multiple data IDs in parallel across multiple CPUs of one machine if you provide a :option:`-j`/:option:`--processes` argument with a value greater than ``1``.
+For example, to spread processing across up to eight CPUs:
+
+.. code-block:: bash
+
+   task.py REPOPATH --output output --id -j 8 ...
+
+When operating with :option:`-j`, command-line tasks have a default per-process timeout of 720 hours.
+You can change this timeout via the :option:`--timeout` command-line argument.
+
+Log messages from each process are combined into the same console output stream.
+To discern logging statements from each process use the :option:`--longlog` argument to enable a verbose logging format that includes the data ID being processed.
+See :ref:`command-line-task-logging-howto-longlog` for more information.
+
+.. note::
+
+   Internally, :option:`-j`-type multiprocessing is implemented with the `multiprocessing` module in the Python standard library.
+
+   If a task does not support :option:`-j` parallel processing it will report a non-fatal warning:
+
+   .. code-block:: text
+
+      This task does not support multiprocessing; using one process
+
+.. _command-line-task-parallel-howto-distributed:
+
+How to enable distributed processing
+====================================
+
+Command-line tasks implemented in the ``pipe_drivers`` package can distribute processing across multiple nodes using MPI or batch submission to clusters (PBS or SLURM).
+See the ``pipe_drivers`` documentation for more information.
+
```

Comment:
It might be worth mentioning here that you can disable this multithreading with `--batchType None`;  then people can at least get started before the documentation exists and Meredith is happy.